### DISCUSSION TITLE
How to enable Voodoo 3?

### DISCUSSION THREAD

**Discussion by Vort** [bochs-emu/Bochs#120] [Q&A] [answered]
Can anyone provide example of configuration file, which really enables Voodoo 3?
I tried different options, but result was the same: `Experimental 3dfx Voodoo Graphics (SST-1/2)` was loaded.
Only when I commented out lines 143, 147, 148, 149 and 150 in `voodoo.cc`, `Experimental 3dfx Voodoo 3 PCI` started to load.

**Comment by vruppert** [comment, marked as answer]
To enable a Voodoo model with VGA subsystem you need to specify the VGA extension option:
vga: extension=voodoo

Then you need to set up the Voodoo model:
voodoo: model=voodoo3

If you chose Voodoo1 or Voodoo2 here, you'll get the selected extension card and VGA-only adapter without additional features.
If the VGA extension is set to another VGA-compatible card (vbe, cirrus), the choices "banshee" and "voodoo3" are not valid.

**Reply by Vort** [reply]
@vruppert do you see `Experimental 3dfx Voodoo 3 PCI` in logs with such config?
I tried and it does not work for me - `Experimental 3dfx Voodoo Graphics (SST-1/2)` appears instead.

**Comment by vruppert** [comment]
Oops, one more setting is important here, since Banshee and Voodoo3 require a specific VGABIOS:
vgaromimag: file VGABIOS-lgpl-latest-banshee

**Reply by Vort** [reply]
Then in addition I have `3dfx Voodoo Banshee-compatible VGA not detected` message.
Ok, looks like Voodoo 3 support is just broken.
Should I make Issue with this problem or this Discussion is enough?

**Reply by vruppert** [reply]
I just tried a Voodooe setup. This is a piece of the log:

00000000000i[PLUGIN] init_dev of 'voodoo' plugin device by virtual method
00000000000i[MEM0  ] Register memory access handlers: 0x0000000a0000 - 0x0000000bffff
00000000000i[VVGA  ] interval=200000, mode=realtime
00000000000i[VVGA  ] VSYNC using realtime mode
00000000000i[DEV   ] PCI slot #1 used by plugin 'voodoo'
00000000000i[DEV   ] Experimental 3dfx Voodoo 3 PCI present at device 2, function 0
00000000000i[VOODOO] loaded PCI ROM '/home/volker/bochs-test/bios/VGABIOS-lgpl-latest-banshee' (size=32768 / PCI=32k)
00000000000i[VOODOO] 3dfx Voodoo Graphics adapter (model=voodoo3) initialized

**Reply by Vort** [reply]
I have found where differences were.
My config also had plugin enabled: `plugin_ctrl: voodoo=true`.
Was this wrong? Can Bochs warn user about it?
It was natural to me to enable plugin before using it.

**Reply by vruppert** [reply]
A Voodoo card with VGA support should not be enabled there. The Bochs plugins version panics in that case, since a plugin cannot be loaded twice. Unfortunately the standard Bochs version without plugins exactly does what you have described.
I have to investigate this issue.

**Reply by vruppert** [reply]
Update: after my latest change the panic should appear similar to plugin version.

**Reply by Vort** [reply]
Ok, thanks. Will check it after fresh problem will be fixed.
Looks like some new problem was introduced during last 2 days.

**Reply by Vort** [reply]
Fresh problem turned out not so easy.
So I tested plugin load right now and I confirm that double plugin load now produces `>>PANIC<< plugin 'voodoo' already loaded`. Voodoo 3 enabling problem is solved now.
